dockerversion/useragent.go

```diff
@@ -5,11 +5,23 @@ import (
 
 	"github.com/docker/docker/pkg/parsers/kernel"
 	"github.com/docker/docker/pkg/useragent"
+	"golang.org/x/net/context"
 )
 
+// uaStringKey is used as key type for user agent string in net/context struct
+type uaStringKey int
```

Comment:
I used `type customType int` approach because it's what's being prescribed in this [Go blog article](https://blog.golang.org/context).